src/mon/MDSMonitor.cc

```diff
@@ -2296,10 +2296,38 @@ void MDSMonitor::tick()
   if (!is_active() || !is_leader()) return;
 
   auto &pending = get_pending_fsmap_writeable();
+  auto now = clock::now();
 
   bool do_propose = false;
   bool propose_osdmap = false;
 
+  if (check_fsmap_struct_version) {
+    /* Allow time for trimming otherwise PaxosService::is_writeable will always
+     * be false.
+     */
+
+    auto since = std::chrono::duration<double>(now-last_fsmap_struct_flush);
+    if (since.count() > 30.0) {
```

Comment:
could just put
```c++
const auto now = clock::now();
const auto elapsed = now - last_fsmap_struct_flush;
if (elapsed > std::chrono::seconds(30)) {
  // ...
}
```

for two reasons:

1. no need to cast the duration to `std::chrono::duration<double>`.
2. s/since/elapsed/`, as, IMHO, `since` is actually `last_fsmap_struct_flush`. what we want to compare is the time elapsed since "`since`".

nit, might want to drop the empty line after the comment. as i think the comment document the intention of the whole `if` block, and the `if (since.count() > 30.0) {` clause in particular, so it might be better to put it a little bit closer to the documented code.

Reply:
Thanks, fixed.